components/brave_wallet/browser/zcash/zcash_rpc.cc

```diff
@@ -350,4 +375,58 @@ void ZCashRpc::OnGetTransactionResponse(
   std::move(callback).Run(response);
 }
 
-}  // namespace brave_wallet::zcash_rpc
+void ZCashRpc::SendTransaction(const std::string& chain_id,
+                               const std::string& data,
+                               SendTransactionCallback callback) {
+  GURL request_url = MakeSendTransactionURL(
+      GetNetworkURL(prefs_, chain_id, mojom::CoinType::ZEC));
+
+  if (!request_url.is_valid()) {
+    std::move(callback).Run(base::unexpected("Request URL is invalid."));
+    return;
+  }
+
+  auto url_loader =
+      MakeGRPCLoader(request_url, MakeSendTransactionParams(data));
+
+  UrlLoadersList::iterator it = url_loaders_list_.insert(
+      url_loaders_list_.begin(), std::move(url_loader));
+
+  (*it)->DownloadToString(
+      url_loader_factory_.get(),
+      base::BindOnce(&ZCashRpc::OnSendTransactionResponse,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback), it),
+      5000);
+}
+
+void ZCashRpc::OnSendTransactionResponse(
```

Comment:
the grammar of this RPC call looks too complex to be parsing it within an unsafe language. Can we make sure we're adhering with the [rule of 2 here](https://chromium.googlesource.com/chromium/src/+/master/docs/security/rule-of-2.md). Ideally, refactoring this to just use api_request_helper would be the easier way since it handles all of the unsafe parsing for us, but I see this isn't relying on a JSON response body, so we may need to handle this in a more specific way.

Reply:
Yup, good point. I'm okay with that for now.

Reply:
https://github.com/brave/brave-browser/issues/34561